packages/react/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright (c) 2023 Gemeente Amsterdam
+ */
+
+import { ChevronLeftIcon, ChevronRightIcon } from '@amsterdam/design-system-react-icons'
+import clsx from 'clsx'
+import { ForwardedRef, forwardRef, HTMLAttributes, useEffect, useState } from 'react'
+import { Icon } from '../Icon/Icon'
+
+export interface PaginationProps extends HTMLAttributes<HTMLElement> {
+  /**
+   * The number of items in your paginated collection.
+   *
+   * Note that this is not the same as the number of pages, but will be used to calculate it.
+   */
+  collectionSize: number
+  /**
+   * Callback triggered when interaction changes the page number.
+   */
+  // eslint-disable-next-line no-unused-vars
+  onPageChange?: (page: number) => void
+  /**
+   * The current page number.
+   */
+  page?: number
+  /**
+   * The number of items per page.
+   */
+  pageSize?: number
```

Comment:
If this needed to calculate the number of pages, shouldn't it be required? If it is not provided it defaults to 10. Then it could be you request the wrong page. Would strong opt for a totalPages prop